docs/reference.md

```diff
@@ -700,43 +703,77 @@ $scope.authObj.$signInWithPopup("google").then(function(result) {
+Authenticates the client using a redirect-based OAuth flow. This function takes a single argument: a
+string or provider object representing the OAuth provider to authenticate with. It returns a
+rejected promise with an `Error` object if the authentication attempt fails. Upon successful
+authentication, the browser will be redirected as part of the OAuth authentication flow. As such,
+the returned promise will never be fulfilled. Instead, you should use the `$onAuthStateChanged()`
+method to detect when the authentication has been successfully completed.
 
-Optionally, you can pass a provider object (like `new firebase.auth().GoogleProvider()`, etc)
-which can be configured with additional options.
+Valid values for the string version of the argument are `"facebook"`, `"github"`, `"google"`, and
+`"twitter"`:
 
 ```js
 $scope.authObj.$signInWithRedirect("google").then(function() {
   // Never called because of page redirect
+  // Instead, use $onAuthStateChanged() to detect successful authentication
 }).catch(function(error) {
   console.error("Authentication failed:", error);
 });
 ```
 
-This method returns a rejected promise with an `Error` object if the authentication attempt fails.
-Upon successful authentication, the browser will be redirected as part of the OAuth authentication
-flow. As such, the returned promise will never be fulfilled. Instead, you should use the `$onAuthStateChanged()`
-method to detect when the authentication has been successfully completed.
+Alternatively, you can request certain scopes or custom parameters from the OAuth provider by
+passing a provider object (such as `new firebase.auth.GoogleAuthProvider()`) configured with
+additional options:
+
+```js
+var provider = new firebase.auth.GoogleAuthProvider();
+provider.addScope("https://www.googleapis.com/auth/plus.login");
+provider.setCustomParameters({
+  login_hint: "user@example.com"
+});
+
+$scope.authObj.$signInWithRedirect(provider).then(function(result) {
+  // Never called because of page redirect
+  // Instead, use $onAuthStateChanged() to detect successful authentication
+}).catch(function(error) {
+  console.error("Authentication failed:", error);
+});
+```
 
-Firebase currently supports Facebook, GitHub, Google, and Twitter authentication. Refer to
+Firebase currently supports Facebook, GitHub, Google, and Twitter authentication. Refer to the
 [authentication documentation](https://firebase.google.com/docs/auth/)
 for information about configuring each provider.
 
 ### $signInWithCredential(credential)
 
-Authenticates the client using a credential (potentially created from OAuth Tokens). This function takes one
-arguments: the credential object. This may be obtained from individual auth providers under `firebase.auth()`;
+Authenticates the client using a credential (potentially created from OAuth tokens). This function
+takes a single argument: the credential object. This may be obtained from individual auth providers
```

Comment:
This sentence doesn't really help me understand how to create a credential object and I think it's going to be easy to miss that `firebase.auth()` refers to the underlying SDK.

Sadly, there doesn't seem to be a good example in the [auth guide](https://firebase.google.com/docs/auth/web/manage-users) or the [api reference](https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signInWithCredential) to help clear this up. So maybe ignore unless you have ideas.